bluemira/base/builder.py

```diff
@@ -103,14 +103,14 @@ class Builder(abc.ABC):
     def __init__(
         self,
         params: Union[Dict, ParameterFrame, ConfigParams, None],
-        build_config: Dict,
+        build_config: Optional[Dict] = None,
     ):
         super().__init__()
         self.name = build_config.get(
             "name", _remove_suffix(self.__class__.__name__, "Builder")
```

Comment:
This will fail as soon as `build_config = None`. Needs some catching and probably a test